cohortextractor/query_language.py

```diff
@@ -215,3 +215,11 @@ class ValueFromCategory(Value):
     def __init__(self, definitions, default):
         self.default = default
         self.definitions = definitions
+
+
+class Codelist(QueryNode):
+    def __init__(self, codes, system, has_categories=False):
+        if has_categories:
+            raise NotImplementedError("categorised codelists are currently unsupported")
+        self.codes = list(codes)
+        self.system = system
```

Comment:
What does `system` do?  I vaguely think I remember you saying it wasn't really necessary for it to be specified in the study definition

Reply:
Clinical codes belong to a coding system, which provides the set of valid codes, their clinical definitions and their relationships. The common ones we use are SNOMED, CTV3 (also known as Read v3), and ICD-10.

Although it isn't strictly necessary for running queries, we really ought to be validating that the codes a user supplies via a codelist match the coding system of the column they're querying against.

At the moment that involves specifying the coding system inline in the study definition. I think what you might be remembering is the [proposal](https://github.com/opensafely-core/cohort-extractor-v2/issues/31) I made recently to grab that information from the opencodelists site and not require the user to specify it manually. I still think we should do that, but it's extra work so for I now I was just going to copy what we do already, as far as possible.

We're also going to need to add metadata to the schema definition so we can record the coding system used for each column that contains codes. Hopefully that's going to be fairly easy to do though.

Reply:
OK, cool.  Sorry, I knew what the `system` was, just not what the purpose was in specifying it here.  Was just wondering if we could remove the requirement to specify it now